demos/coreHTTP/http_demo_s3_download_multithreaded.c

```diff
@@ -198,31 +213,34 @@ typedef struct RequestItem
 /**
  * @brief Data type for response queue.
  *
- * In addition to sending the #HTTPResponse_t, we need to send the buffer it
- * uses as the pointer to the buffer in #HTTPRequestHeaders_t may not be
- * accurate after being copied into the queue.
+ * Contains the response data type and its corresponding buffer, to be read from
+ * the response queue by the main thread.
  */
 typedef struct ResponseItem
 {
     HTTPResponse_t xResponse;
     uint8_t ucResponseBuffer[ democonfigUSER_BUFFER_LENGTH ];
 } ResponseItem_t;
 
-/*
+/**
  * @brief Struct used for sending requests to the HTTP thread.
  *
- * This is used by both the main and HTTP threads. The global scope ensures that
- * it will be located at the same address in the main and HTTP threads, so that
- * pointers remain valid when copied over.
+ * This is modified by the main task and accessed by the HTTP task. The global
+ * scope ensures that it will be located at the same address in the main and
+ * HTTP tasks, so that pointers remain valid when copied over. Once it is placed
+ * on the queue by the main task, it is modified only after the HTTP task has
+ * successfully sent the HTTP request.
```

Comment:
I think it will be better to explicitly mention that there are no other synchronization primitives used.